Microsoft.Identity.Web/Client/TokenAcquisition.cs

```diff
@@ -266,6 +266,23 @@ public async Task RemoveAccount(RedirectContext context)
             }
         }
 
+        IConfidentialClientApplication application;
+
+        /// <summary>
+        /// Creates an MSAL Confidential client application if needed
+        /// </summary>
+        /// <param name="httpContext"></param>
+        /// <param name="claimsPrincipal"></param>
+        /// <returns></returns>
+        private IConfidentialClientApplication GetOrBuildConfidentialClientApplication(HttpContext httpContext, ClaimsPrincipal claimsPrincipal)
```

Comment:
With this as a singleton, how do we handle different user token caches on each request?  Will this sample provide guidance on how to use this with multiple users?

Reply:
this does work with multiple users, @MarkZuber as:
- the code is called in the context of an Identity (.User). ASP.NET core takes care of this
- the cache is deserialized overriding the previous content. The key of each cache is the MSA unique id <tenantId>.<userId>

Do I miss something?

Reply:
sorry not sure if I am following, the singleton application instance will be shared across all user requests, isn't that true?

Reply:
@bryanchen-d 
I've taken this offline with @MarkZuber 
the synchronization is fine (MSAL does the right thing). The only concern with using a singleton is that the cache serialization operations will be a bottleneck, therefore affecting performance

Moving the ITokenAcquisition service back to be scoped (that is per request). The same will be used for the same user, but different users will have different services, which will make things scale better.

cc: @kalyankrishna1 @TiagoBrenck